kanidm_tools/src/cli/session.rs

```diff
@@ -142,11 +142,18 @@ impl LoginOpt {
         self.copt.debug
     }
 
-    fn do_password(&self, client: &mut KanidmClient) -> Result<AuthResponse, ClientError> {
-        let password = rpassword::prompt_password_stderr("Enter password: ").unwrap_or_else(|e| {
-            error!("Failed to create password prompt -- {:?}", e);
-            std::process::exit(1);
-        });
+    fn do_password(
+        &self,
+        client: &mut KanidmClient,
+        pass: Option<String>,
+    ) -> Result<AuthResponse, ClientError> {
+        let password = match pass {
```

Comment:
I feel like the `pass: Option<>` is a bit weird, it might be better to have a way to have a trait or a call ack that defines a way to get the password, which can then hook back to the relevant class of ui.

Reply:
Ok, I'll revert backs all change to cli libs and start using kanidm_client ...is it ok?

Reply:
It is done now TUI does not use cli session.

Reply:
Password is not option anymore.